torchaudio/prototype/rnnt_loss.py

```diff
@@ -166,17 +105,30 @@ def rnnt_loss(
             False  # softmax needs the original logits value
         )
 
-    cost = _RNNT.apply(
-        logits,
-        targets,
-        logit_lengths,
-        target_lengths,
-        blank,
-        clamp,
-        fused_log_softmax,
-        reuse_logits_for_grads,
-    )
-    return cost
+    # move everything to the same device.
+    targets = targets.to(device=logits.device)
+    logit_lengths = logit_lengths.to(device=logits.device)
+    target_lengths = target_lengths.to(device=logits.device)
```

Comment:
i can see a situation where a user moved all but the logits accidentally, and would encounter a very slow computation of the loss function. do we do this elsewhere in the codebase? were the checks in the c++ code not catching this already and raising an error?